go.mod

```diff
@@ -1,8 +1,8 @@
 module github.com/openshift/machine-api-provider-nutanix
 
-go 1.22.5
+go 1.23.0
 
-toolchain go1.22.8
+toolchain go1.23.3
 
```

Comment:
We can't bump to `go 1.23` in release-4.18 unfortunately.

I suggest reading: https://redhat-internal.slack.com/archives/CE4UC6AVB/p1741170208238289
To find a solution on how to do this bump.

Reply:
@damdo Hi, the slack discussion is very long, can you let me know what is the right way to fix this CVE without changing the go/toolchain version to 1.23?

Reply:
@yanhua121  I sent you a DM on Slack

Reply:
Yes, got it. And made the changes based the instructions you provided. Please take another look.